# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmValType.cpp
# Commit: a654205d8470
# Full Hash: a654205d84701ba611f22f0020be612e0db887c6
# Author: Ben Visness <bvisness@mozilla.com>
# Date: 2023-09-14 04:15:24
# Description:
#   Bug 1851569: Add i31 to ref.cast test suite. r=jseward
#   
#   This thoroughly tests i31's interaction with other types, and fixes a couple subtle bugs. For example, casting an anyref to an eqref was incorrectly failing if the value was i31.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D188108
# ==============================================================================

diff -r 893bba06f7dd -r a654205d8470 js/src/wasm/WasmValType.cpp
--- a/js/src/wasm/WasmValType.cpp	Wed Sep 13 20:28:18 2023 +0000
+++ b/js/src/wasm/WasmValType.cpp	Wed Sep 13 20:57:56 2023 +0000
@@ -352,7 +352,7 @@
       heapType = "eq";
       break;
     case RefType::I31:
-      heapType = "eq";
+      heapType = "i31";
       break;
     case RefType::Struct:
       heapType = "struct";